### PR TITLE
Document Mappings Source Extensions

### PR DESCRIPTION
**Pull request by bharatnpti**
## References

- TODO

## Submitter checklist

- [ x] The PR request is well described and justified, including the body and the references
- [ x] The PR title represents the desired changelog entry
- [ x] If the change against WireMock 2 functionality (incompatible with WireMock 3),
      it is submitted against the [2.x](https://github.com/wiremock/wiremock.org/tree/2.x) branch
- [ x] The repository's code style is followed (see the contributing guide)

### CODE REVIEW

File: _docs/extensibility/adding-mappings-source.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+description: Adding additional source of loading mappings via extensions.
+---
+
+Additional source to load the stub mappings can be configured by implementing `MappingsSourceExtension`.
+
+```java
+public class DummyMappingsSourceExtension extends MappingsSourceExtension {
+  @Override
+  public String getName() {
+    return null; // Return the name of extension
```
Comment (tomakehurst) [inline _docs/extensibility/adding-mappings-source.md:14]: People will inevitably copy-paste examples verbatim so please can we change this to a non-null String.

File: _docs/extensibility/adding-mappings-source.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  }
+
+  @Override
+  public void loadMappingsInto(StubMappings stubMappings) {
+    {
```
Comment (tomakehurst) [inline _docs/extensibility/adding-mappings-source.md:19]: Seem to be some redundant braces here.

File: _docs/extensibility/adding-mappings-source.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        // mappings can be loaded from any source like git repo, database, file storage, stc
+    }
+  }
+}
+```
+
+Registering the extension with wiremock.
+
+```java
+public class Main {
```
Comment (tomakehurst) [inline _docs/extensibility/adding-mappings-source.md:30]: Suggest removing all the main class stuff and just show newing up `WireMockServer`. This is how the other examples are done.

**Review by tomakehurst — CHANGES_REQUESTED** [review]
A couple more things:
1. You'll need to add the new page to `doc-categories.yml` for it to be added to the navigation.
2. Please could you add a bullet point and link to this page from `extending-wiremock.md` which is the overview page.

File: _docs/extensibility/adding-mappings-source.md (hunk 3 above)
Comment (bharatnpti) [reply _docs/extensibility/adding-mappings-source.md]: All 5 feedback point incorporated

**Comment by bharatnpti** [comment]
Hi @tomakehurst,

I have raised another PR including the suggested changes - https://github.com/wiremock/wiremock.org/pull/181

**Pull request closed without merging**